Mathlib/Data/Nat/Factorial/SuperFactorial.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2023 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Data.Nat.Factorial.Basic
+
+/-!
+# Super factorial
+
+This file defines the super factorial `1! * 2! * 3! * ...* n!`.
+
+## Main declarations
+
+* `Nat.superFactorial`: The super factorial.
+-/
+
+
+namespace Nat
+
+/-- `Nat.superFactorial n` is the super factorial of `n`. -/
+@[simp]
+def superFactorial : ℕ → ℕ
+  | 0 => 1
+  | succ n => factorial n.succ * superFactorial n
+
+/-- `sf` notation for super factorial -/
+scoped notation  "sf" n => Nat.superFactorial n
+
+section SuperFactorial
+
+variable {n : ℕ}
+
+@[simp]
+theorem superFactorial_zero : (sf 0) = 1 :=
```

Comment:
Are the `()`s needed here? If so, I think you need to adjust the precedence.

Reply:
Ok, will adjust the precedence

Reply:
When `(sf n)` is left of the `=` the `()`s seem to be needed and I didn't manage to adjust the precedence to fix this.

Reply:
ok, meanwhile, I fixed the precende, not entirely sure if "60" is a good value, but with that no `()`s are needed anymore.